clickhouse/columns/array.cpp

```diff
@@ -78,6 +78,13 @@ size_t ColumnArray::Size() const {
     return offsets_->Size();
 }
 
+void ColumnArray::Swap(Column& other) {
+    if (auto col = dynamic_cast<ColumnArray*>(&other)) {
```

Comment:
For this and all other `Swap()`'s where the dynamic cast is performed, is it a correct assumption to make, that nothing should happen if the actual type of the argument doesn't match the `*this` type?

Reply:
Maybe a dynamic casting of the reference, with a possibility of an exception, is a safer way to do this? (Unless there is a proper way of handling such situations.)

Reply:
Well, in the perfect world that would throw an exception, but right now it is consistent with the rest of the code. I assume one day we'll implement proper error management.

Reply:
I think it's better to change in this PR, because it looks like there is no `dynamic_cast` anywhere else. The change is pretty mechanic though:

Reply:
Yeah, except for `std::dynamic_pointer_cast` [here](https://github.com/ClickHouse/clickhouse-cpp/blob/master/clickhouse/columns/column.h#L23), which is used ALOT.

But I agree, let's change it here, in `Swap()` that is the right thing to do. The rest is coming later.

Reply:
done